polkadot/node/core/chain-selection/src/db_backend/v1.rs

```diff
@@ -246,7 +246,7 @@ impl Backend for DbBackend {
 			})
 			.enumerate()
 			.take_while(|(idx, r)| {
-				r.as_ref().map_or(true, |(at, _)| *at <= up_to.into() && *idx < max_elements)
+				r.as_ref().map_or(true, |(at, _)| *at <= up_to as u64 && *idx < max_elements)
```

Comment:
`as` is less safe than explicit conversion because it can truncate the value.

`up_to` is already a `u64`.
```suggestion
				r.as_ref().map_or(true, |(at, _)| *at <= up_to && *idx < max_elements)
```

Reply:
Good catch! Thanks.